src/Pages/yoojinkwon/Login/Login.js

```diff
@@ -1,13 +1,56 @@
 import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+import './Login.scss';
+
+class LoginYoojin extends Component {
+    constructor(){
+        super();
+        this.state = {
+            loginId : " ",
+            loginPw : " ",
+        };
+    }
+
+    goToMain = () => {
+        this.props.history.push('/main-yoojin');
+    }  
+
+    handleInput = (e) => {
+        this.setState({
+            [e.target.name] : e.target.value,
+        })
+    }
 
-class Login extends Component {
     render() {
+        const validation = this.state.loginId.includes("@") && this.state.loginPw.length > 4
```

Comment:
새로운 state를 생성하지 않고, 기존 state들로부터 얻을 수 있는 값이라는 점을 잘 활용해 주셨네요👍
하지만 유지보수, 가독성을 위해서 validation 실제 로직 부분을 함수로 따로 분리하면 더 좋을 것 같습니다.
```suggestion
        const isValid = this.isValidId() && this.isValidPw()
        // 혹은 validation 함수를 하나로 묶어서
        // const isValid = this.validateFormInput()
```

예를 들어 `isValidId()`의 경우 현재 state의 `loginId`값으로부터 적절한 Id 인지 판단하여 true 인지 false 인지 반환이 되겠죠? 위 수정된 코드는 가독성이 좋습니다. 마치 책을 읽듯이
**valid한 아이디와 valid한 패스워드 모두 만족시킬 때 isValid가 true가 되는구나**
라는 생각이 들겠죠. Id validation 로직과 pw validation 로직이 복잡해질 경우(실제로 웹 사이트의 패스워드 검증 로직은 복잡하다는 것을 잘 알고 계실 겁니다) 분리된 함수에서 로직만 변경하면 될 것입니다.

물론! 작성해주신 코드와 같이 간단한 로직의 경우 가독성이 떨어진다고 볼 순 없고, 충분히 쉽게 이해할 수 있는 로직이긴 합니다. 제가 드린 리뷰는 학습과 추후 확장성을 위해 남겨드린 리뷰이며, 작성해주신 코드와 같은 id, pw 검증 로직을 사용한다면 저는 그대로 두어도 크게 문제되지 않는다고 생각합니다.

Reply:
네 장현님! 로직이 복잡해진다면 훨씬 가독성이 좋아질 것 같습니다. 수정 시도해보겠습니다💪